adobe/spectrum-web-components
# fix(menu): prevent sp-menu-item text-align cascading

## Description

Previous to the Overlay V2, the `text-align` property of action-menu, or similar components, did not cascade into the `sp-menu-items`. This PR brings back that behavior by setting the `text-align: initial` on `sp-menu-item`. 

PS: This is the same behavior as in React Spectrum. As per Spectrum guidelines I haven't yet seen a use-case for other types of aligment. If they are needed, we can provide a `--mod-*` type css variable.
There is also an on-going discussion on a spectrum-css [issue](https://github.com/adobe/spectrum-css/issues/3058) but this missalignment in behavior comes from the SWC's approach to Overlay so I think this is a good place to fix.


## Related issue(s)

- N/A

## Motivation and context

Fixes the text-align cascading into the overlay menu-items.

## How has this been tested?




    1. Open [ActionMenu Storybook](https://imizga-text-align--spectrum-web-components.netlify.app/storybook/?path=/story/action-menu--default)
    2. Place a `text-align: center` on a parent of the ActionMenu (e.g. sp-theme) 
    3. Click the ActionMenu
    4. Observe that the text inside the menu-items is centered


## Screenshots (if appropriate)

Without the change:
<img width="1095" alt="without the fix" src="https://github.com/user-attachments/assets/0b2f62c0-a41e-4aeb-bc5c-28047065e0a7">


After the change:
<img width="1480" alt="with the fix" src="https://github.com/user-attachments/assets/d1d54c9c-2b91-4153-aadb-49457bbf6c2a">


## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
